lib/task_helpers/exports/tags.rb

```diff
@@ -0,0 +1,36 @@
+module TaskHelpers
+  class Exports
+    class Tags
+      # Description attribute of Tag Categories that are not visible in the UI
+      SPECIAL_TAGS = ['Parent Folder Path (VMs & Templates)', 'Parent Folder Path (Hosts & Clusters)', 'User roles'].freeze
+
+      def export(options = {})
+        export_dir = options[:directory]
+
+        tags = if options[:all]
+                 Classification.where('parent_id = ? AND description NOT IN (?)', 0, SPECIAL_TAGS)
+               else
+                 export_tags = []
+                 Classification.where('parent_id = ? AND description NOT IN (?)', 0, SPECIAL_TAGS).each do |cat|
+                   if !cat.default
+                     export_tags << cat
+                   else
+                     cat.entries.each do |entry|
+                       unless entry.default
+                         export_tags << cat
+                         break
```

Comment:
Not sure why there's a `break` here. Don't you need to check the rest of the entries to get any others that are not "default"?

Reply:
Once one non-defualt entry is found the category will be exported which will include all entries.

I could adjust the export to only ever export non-default entries, but it is possible to change the name and description so I felt it was best to export all entries if one non-default was found. Granted there is an edge case that isn't covered where a name or description of a default is changed and no entries are added. In that case the category will never be exported.

This logic is a little different than the script in rhtconsulting which exports all but the "special" categories.